pennylane/devices/experimental/default_qubit_2.py

```diff
@@ -215,9 +267,91 @@ def compute_derivatives(
             self.tracker.record()
 
         if execution_config.gradient_method == "adjoint":
-            res = tuple(adjoint_jacobian(circuit) for circuit in circuits)
+            max_workers = self._get_max_workers(execution_config)
+            if max_workers is None:
+                res = tuple(adjoint_jacobian(circuit) for circuit in circuits)
+            else:
+                vanilla_circuits = [convert_to_numpy_parameters(c) for c in circuits]
+                with concurrent.futures.ProcessPoolExecutor(max_workers=max_workers) as executor:
+                    exec_map = executor.map(adjoint_jacobian, vanilla_circuits)
+                    res = tuple(circuit for circuit in exec_map)
+
+                # reset _rng to mimic serial behavior
+                self._rng = np.random.default_rng(self._rng.integers(2**31 - 1))
+
             return res[0] if is_single_circuit else res
 
         raise NotImplementedError(
             f"{self.name} cannot compute derivatives via {execution_config.gradient_method}"
         )
+
+    # pylint: disable=missing-function-docstring
+    def _get_max_workers(self, execution_config=None):
+        max_workers = None
+        if execution_config is not None:
```

Comment:
```suggestion
        if execution_config:
```
Can treat `None` as falsy.